docs/containers-registries.conf.5.md

```diff
@@ -18,80 +18,141 @@ VERSION 2 is the latest format of the `registries.conf` and is currently in
 for now.
 
-Every registry can have its own mirrors configured.  The mirrors will be tested
-in order for the availability of the remote manifest.  This happens currently
-only during an image pull.  If the manifest is not reachable due to connectivity
-issues or the unavailability of the remote manifest, then the next mirror will
-be tested instead.  If no mirror is configured or contains the manifest to be
-pulled, then the initially provided reference will be used as fallback.  It is
-possible to set the `insecure` option per mirror, too.
-
-Furthermore it is possible to specify a `prefix` for a registry.  The `prefix`
-is used to find the relevant target registry from where the image has to be
-pulled.  During the test for the availability of the image, the prefixed
-location will be rewritten to the correct remote location.  This applies to
-mirrors as well as the fallback `location`.  If no prefix is specified, it
-defaults to the specified `location`.  For example, if
-`prefix = "example.com/foo"`, `location = "example.com"` and the image will be
-pulled from `example.com/foo/image`, then the resulting pull will be effectively
-point to `example.com/image`.
-
-By default container runtimes use TLS when retrieving images from a registry.
-If the registry is not setup with TLS, then the container runtime will fail to
-pull images from the registry. If you set `insecure = true` for a registry or a
-mirror you overwrite the `insecure` flag for that specific entry.  This means
-that the container runtime will attempt use unencrypted HTTP to pull the image.
-It also allows you to pull from a registry with self-signed certificates.
-
-If `blocked = true` then it is not allowed to pull images from that registry.
-
-If `mirror-by-digest-only = true`, mirrors will only be used during pulling if
-the image is referred to by digest.  Referencing an image by digest enforces to
-always return the same image, independent if it is being pulled from the main
-registry or a mirror.  Pulling an image referenced by tag could yield different
-results, depending on which mirror or registry it is being pulled from.  Setting
-this attribute makes sure to avoid such a situation but requires the primary
-registry to always be available if references by tag are used.
-
-In addition, at the top level, an `unqualified-search-registries` array can be
-specified; this is a list of host[:port] (not more general prefix!) entries to
-use when pulling an unqualified image, in order.
+### GLOBAL SETTINGS
+
+`unqualified-search-registries`
+: An array of host[:port] (not more general prefix!) entries to use when pulling
+an unqualified image, in order.
+
+### NAMESPACED `[[registry]]` SETTINGS
+
+The following settings may differ among different registries, and namespaces/repositories
+within a registry.  The configuration is represented as a sequence of `[[registry]]`
+entries containging the specific settings.
```

Comment:
Non-blocking: This description seems specific to the implications of the `prefix` which may confuse readers. I'd prefer something more general describing that registries (and namespaces) can be configured in a TOML table array, each entry starting with `[[registry]]`. Then we could merge the following sections and go through the fields and sub-tables one after another.

Reply:
Reworded to start with saying that there is an array of `[[registry]]` TOML tables, and to let the “Choosing a [[registry]] TOML table” section speak for itself.

For now I’m leaving the “choosing table” / “misc. settings” / “remapping and mirroring” sections separate, because the “choosing table” one must be understood by everyone, while the other two are optional — and in particular the whole concept of remapping/mirroring can be skipped by users who don’t need that; mixing `location` among all the other options without further context seems confusing to me.